### ISSUE TITLE
Add column throws exception after adding column

### ISSUE THREAD

**Issue by wittlesouth** [sqlalchemy/alembic#589]
Environment:

Server version: 5.7.26 MySQL Community Server (GPL)
SQLAlchemy==1.2.17
mysqlclient==1.3.14
alembic==1.0.11

I did modify env.py to pull the connect string from an environment variable, and am calling create_engine directly. The connect string is constructed with the following code, where the format parameters come from environment variables:

```
    connect_string = "mysql+mysqldb://{}:{}@{}:{}/{}?charset=utf8mb4&binary_prefix=True".format(\
                 DB_USER, DB_PWD, DB_HOST, DB_PORT, DB_NAME)
    if connect_string:
        connectable = create_engine(connect_string)
```

I'm trying to add a JSON column to a table named User. The version python file has this for upgrade:

```
def upgrade():
    op.add_column('User', sa.Column('profiles', sa.String(20)))
```

User table before running alembic:

```
mysql> describe User;
+------------------+--------------+------+-----+-------------------+-------+
| Field            | Type         | Null | Key | Default           | Extra |
+------------------+--------------+------+-----+-------------------+-------+
| record_created   | datetime     | YES  |     | CURRENT_TIMESTAMP |       |
| record_updated   | datetime     | YES  |     | CURRENT_TIMESTAMP |       |
| _creator_user_id | binary(16)   | YES  |     | NULL              |       |
| user_id          | binary(16)   | NO   | PRI | NULL              |       |
| username         | varchar(32)  | YES  |     | NULL              |       |
| email            | varchar(80)  | YES  |     | NULL              |       |
| first_name       | varchar(80)  | YES  |     | NULL              |       |
| full_name        | varchar(120) | YES  |     | NULL              |       |
| last_name        | varchar(80)  | YES  |     | NULL              |       |
| phone            | varchar(20)  | YES  |     | NULL              |       |
| source           | varchar(80)  | YES  |     | NULL              |       |
| create_users     | tinyint(1)   | YES  |     | NULL              |       |
| create_groups    | tinyint(1)   | YES  |     | NULL              |       |
| grant_privs      | tinyint(1)   | YES  |     | NULL              |       |
| reset_code       | varchar(6)   | YES  |     | NULL              |       |
| reset_expires    | datetime     | YES  |     | NULL              |       |
| password_hash    | varchar(128) | YES  |     | NULL              |       |
+------------------+--------------+------+-----+-------------------+-------+
17 rows in set (0.01 sec)
```

Then I run `alembic upgrade head`, and get the exception stack trace below. Weirdly, though, the exception happens after the table has been successfully altered, here is what the user table looks like once Alembic completes:

```
mysql> describe User;
+------------------+--------------+------+-----+-------------------+-------+
| Field            | Type         | Null | Key | Default           | Extra |
+------------------+--------------+------+-----+-------------------+-------+
| record_created   | datetime     | YES  |     | CURRENT_TIMESTAMP |       |
| record_updated   | datetime     | YES  |     | CURRENT_TIMESTAMP |       |
| _creator_user_id | binary(16)   | YES  |     | NULL              |       |
| user_id          | binary(16)   | NO   | PRI | NULL              |       |
| username         | varchar(32)  | YES  |     | NULL              |       |
| email            | varchar(80)  | YES  |     | NULL              |       |
| first_name       | varchar(80)  | YES  |     | NULL              |       |
| full_name        | varchar(120) | YES  |     | NULL              |       |
| last_name        | varchar(80)  | YES  |     | NULL              |       |
| phone            | varchar(20)  | YES  |     | NULL              |       |
| source           | varchar(80)  | YES  |     | NULL              |       |
| create_users     | tinyint(1)   | YES  |     | NULL              |       |
| create_groups    | tinyint(1)   | YES  |     | NULL              |       |
| grant_privs      | tinyint(1)   | YES  |     | NULL              |       |
| reset_code       | varchar(6)   | YES  |     | NULL              |       |
| reset_expires    | datetime     | YES  |     | NULL              |       |
| password_hash    | varchar(128) | YES  |     | NULL              |       |
| profiles         | json         | YES  |     | NULL              |       |
+------------------+--------------+------+-----+-------------------+-------+
18 rows in set (0.01 sec)
```

You can see that the table was altered as I expected. However, there are other schema operations that should be performed for this version, and they are not executed. Here is the stack trace. 

```
(server) Erics-2018-MBP:server eric$ alembic upgrade head
INFO  [alembic.runtime.migration] Context impl MySQLImpl.
INFO  [alembic.runtime.migration] Will assume non-transactional DDL.
INFO  [alembic.runtime.migration] Running upgrade  -> b423627c6fe0, Add profile
Traceback (most recent call last):
  File "/Users/eric/Development/wsv/muam/server/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 1236, in _execute_context
    cursor, statement, parameters, context
  File "/Users/eric/Development/wsv/muam/server/lib/python3.7/site-packages/sqlalchemy/engine/default.py", line 536, in do_execute
    cursor.execute(statement, parameters)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/MySQLdb/cursors.py", line 255, in execute
    self.errorhandler(self, exc, value)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/MySQLdb/connections.py", line 50, in defaulterrorhandler
    raise errorvalue
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/MySQLdb/cursors.py", line 252, in execute
    res = self._query(query)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/MySQLdb/cursors.py", line 378, in _query
    db.query(q)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/MySQLdb/connections.py", line 280, in query
    _mysql.connection.query(self, query)
_mysql_exceptions.OperationalError: (1033, "Incorrect information in file: './muam/user.frm'")

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/eric/Development/wsv/muam/server/bin/alembic", line 10, in <module>
    sys.exit(main())
  File "/Users/eric/Development/wsv/muam/server/lib/python3.7/site-packages/alembic/config.py", line 540, in main
    CommandLine(prog=prog).main(argv=argv)
  File "/Users/eric/Development/wsv/muam/server/lib/python3.7/site-packages/alembic/config.py", line 534, in main
    self.run_cmd(cfg, options)
  File "/Users/eric/Development/wsv/muam/server/lib/python3.7/site-packages/alembic/config.py", line 514, in run_cmd
    **dict((k, getattr(options, k, None)) for k in kwarg)
  File "/Users/eric/Development/wsv/muam/server/lib/python3.7/site-packages/alembic/command.py", line 276, in upgrade
    script.run_env()
  File "/Users/eric/Development/wsv/muam/server/lib/python3.7/site-packages/alembic/script/base.py", line 475, in run_env
    util.load_python_file(self.dir, "env.py")
  File "/Users/eric/Development/wsv/muam/server/lib/python3.7/site-packages/alembic/util/pyfiles.py", line 90, in load_python_file
    module = load_module_py(module_id, path)
  File "/Users/eric/Development/wsv/muam/server/lib/python3.7/site-packages/alembic/util/compat.py", line 177, in load_module_py
    spec.loader.exec_module(module)
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "muamdb/env.py", line 82, in <module>
    run_migrations_online()
  File "muamdb/env.py", line 76, in run_migrations_online
    context.run_migrations()
  File "<string>", line 8, in run_migrations
  File "/Users/eric/Development/wsv/muam/server/lib/python3.7/site-packages/alembic/runtime/environment.py", line 839, in run_migrations
    self.get_context().run_migrations(**kw)
  File "/Users/eric/Development/wsv/muam/server/lib/python3.7/site-packages/alembic/runtime/migration.py", line 362, in run_migrations
    step.migration_fn(**kw)
  File "/Users/eric/Development/wsv/muam/server/muamdb/versions/b423627c6fe0_add_profile.py", line 18, in upgrade
    op.add_column('User', sa.Column('profiles', sa.JSON))
  File "<string>", line 8, in add_column
  File "<string>", line 3, in add_column
  File "/Users/eric/Development/wsv/muam/server/lib/python3.7/site-packages/alembic/operations/ops.py", line 1904, in add_column
    return operations.invoke(op)
  File "/Users/eric/Development/wsv/muam/server/lib/python3.7/site-packages/alembic/operations/base.py", line 345, in invoke
    return fn(self, operation)
  File "/Users/eric/Development/wsv/muam/server/lib/python3.7/site-packages/alembic/operations/toimpl.py", line 131, in add_column
    operations.impl.add_column(table_name, column, schema=schema)
  File "/Users/eric/Development/wsv/muam/server/lib/python3.7/site-packages/alembic/ddl/impl.py", line 231, in add_column
    self._exec(base.AddColumn(table_name, column, schema=schema))
  File "/Users/eric/Development/wsv/muam/server/lib/python3.7/site-packages/alembic/ddl/impl.py", line 134, in _exec
    return conn.execute(construct, *multiparams, **params)
  File "/Users/eric/Development/wsv/muam/server/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 980, in execute
    return meth(self, multiparams, params)
  File "/Users/eric/Development/wsv/muam/server/lib/python3.7/site-packages/sqlalchemy/sql/ddl.py", line 72, in _execute_on_connection
    return connection._execute_ddl(self, multiparams, params)
  File "/Users/eric/Development/wsv/muam/server/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 1042, in _execute_ddl
    compiled,
  File "/Users/eric/Development/wsv/muam/server/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 1240, in _execute_context
    e, statement, parameters, cursor, context
  File "/Users/eric/Development/wsv/muam/server/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 1458, in _handle_dbapi_exception
    util.raise_from_cause(sqlalchemy_exception, exc_info)
  File "/Users/eric/Development/wsv/muam/server/lib/python3.7/site-packages/sqlalchemy/util/compat.py", line 296, in raise_from_cause
    reraise(type(exception), exception, tb=exc_tb, cause=cause)
  File "/Users/eric/Development/wsv/muam/server/lib/python3.7/site-packages/sqlalchemy/util/compat.py", line 276, in reraise
    raise value.with_traceback(tb)
  File "/Users/eric/Development/wsv/muam/server/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 1236, in _execute_context
    cursor, statement, parameters, context
  File "/Users/eric/Development/wsv/muam/server/lib/python3.7/site-packages/sqlalchemy/engine/default.py", line 536, in do_execute
    cursor.execute(statement, parameters)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/MySQLdb/cursors.py", line 255, in execute
    self.errorhandler(self, exc, value)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/MySQLdb/connections.py", line 50, in defaulterrorhandler
    raise errorvalue
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/MySQLdb/cursors.py", line 252, in execute
    res = self._query(query)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/MySQLdb/cursors.py", line 378, in _query
    db.query(q)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/MySQLdb/connections.py", line 280, in query
    _mysql.connection.query(self, query)
sqlalchemy.exc.OperationalError: (_mysql_exceptions.OperationalError) (1033, "Incorrect information in file: './muam/user.frm'") [SQL: 'ALTER TABLE `User` ADD COLUMN profiles JSON'] (Background on this error at: http://sqlalche.me/e/e3q8)
```

The DDL listed seems fine. Any suggestions as to what I might be doing wrong?

**Comment by wittlesouth** [comment]
OK, never mind, this error is coming from mysql; it turns out the DDL is executing, but MYSQL is responding with this error for some reason.

**Comment by wittlesouth** [comment]
In case anyone finds this later, it appears to be an issue running mysql in a container in a Kubernetes environment where the persistent volume is a VMware Fusion shared folder. When I moved the persistent volume used for the mysql container from a shared folder to the VM local filesystem, the problem went away. Before that, I was able to consistently generate a 1033 error from mysql whenever I ran an alter table statement.
